### PR TITLE
Add `forwarded_kwrestarg` node to `AST::Builder`

### PR DESCRIPTION
**Pull request by koic**
This PR adds `forwarded_kwrestarg` node to `AST::Builder`.

This feature is required to resolve https://github.com/rubocop/rubocop/issues/11730.

I thought about defining a class like `ForwardedKwrestargNode` for `forwarded_kwrestarg`, but the node is actually an anonymous `KeywordSplatNode` so it's used.

### CODE REVIEW

File: lib/rubocop/ast/builder.rb [hunk 1]
```diff
@@ -19,76 +19,77 @@ class Builder < Parser::Builders::Default
...
+        class:               ClassNode,
+        const:               ConstNode,
+        def:                 DefNode,
+        defined?:            DefinedNode,
+        defs:                DefNode,
+        dstr:                DstrNode,
+        ensure:              EnsureNode,
+        for:                 ForNode,
+        forward_args:        ForwardArgsNode,
+        forwarded_kwrestarg: KeywordSplatNode,
```
Comment (koic) [inline lib/rubocop/ast/builder.rb:55]: This line is the only one added in this map.

File: spec/rubocop/ast/keyword_splat_node_spec.rb [hunk 2]
```diff
@@ -39,6 +39,82 @@
...
+    let(:kwsplat_node) { parse_source(source).ast.children.last.children.last }
+
+    describe '.new' do
+      let(:source) do
+        <<~RUBY
+          def foo(**)
+            { a: 1, ** }
+          end
+        RUBY
+      end
```
Comment (marcandre) [inline spec/rubocop/ast/keyword_splat_node_spec.rb:52]: Could we share the `let(:source)`? It would also be closer to the `let(:kwsplat_node)`

Comment (koic) [reply spec/rubocop/ast/keyword_splat_node_spec.rb]: Ah, makes sense! I've updated it. Thank you for the suggestion!

**Comment by marcandre** [comment]
My only hesitation is: do we want a different class, or could we not use `Kwsplat`?

**Comment by koic** [comment]
Ah, yeah, I thought it would be better to be able to distinguish between `kwsplat` node type and `forwarded_kwrestarg` node type, but it seems pragmatic and more better to have compatibility with `kwsplat_type?` as you suggested. I've updated this PR.

File: spec/rubocop/ast/keyword_splat_node_spec.rb [hunk 3]
```diff
@@ -39,6 +39,45 @@
...
+      it { expect(kwsplat_node.value).to eq(kwsplat_node) }
+    end
+
+    describe '#operator' do
+      it { expect(kwsplat_node.operator).to eq('**') }
+    end
+
+    describe '#kwsplat_type?' do
+      it { expect(kwsplat_node).to be_kwsplat_type }
+    end
```
Comment (marcandre) [inline spec/rubocop/ast/keyword_splat_node_spec.rb:78]: Great.
We also need `it { expect(kwsplat_node).to be_forwarded_kwrestarg_type }`, right?

Comment (koic) [reply spec/rubocop/ast/keyword_splat_node_spec.rb:75]: I see! surely! I've add specs for that.

>  distinguish between `kwsplat` node type and `forwarded_kwrestarg` node type

This worry of mine was unfounded 😅

**Comment by marcandre** [comment]
Thank you so much! Released as 1.28.0 🚀

**Pull request merged**